### PR TITLE
Updated API plugin configurations (main)

### PR DESCRIPTION
**Pull request by korydraughn**

### CODE REVIEW

**Review by alanking — COMMENTED** [review]
Do these APIs work when using federation? Does `REMOTE_USER_AUTH` mean that a remote user can authenticate directly with a server in a remote zone and invoke these APIs?

**Comment by korydraughn** [comment]
I haven't tested this yet, but each API plugin in this PR uses the same values that `rxDataObjPut` uses, so there shouldn't be any issues.

We'll know more once I run the tests.

**Comment by korydraughn** [comment]
All unit tests pass.

Running federation tests.

**Comment by korydraughn** [comment]
All federation tests passed.

Running core tests.

**Comment by trel** [comment]
hooray!

**Comment by korydraughn** [comment]
Yes. Will do that soon.

Do we agree with the changes? I'm going to check a few other APIs before we merge.

**Comment by alanking** [comment]
I think I would like it more if we understood the change better. It works with or without the change, so, I would like to know what's changing.

**Comment by korydraughn** [comment]
Let's discuss over zoom today.

**Comment by korydraughn** [comment]
After some discussion, we've decided that these changes need unit tests to assert expected behavior.

File: unit_tests/src/test_rc_data_obj.cpp [hunk 1]
```diff
@@ -1528,3 +1530,23 @@ TEST_CASE("#7250: updates to ticket stats do not hang")
...
+    CHECK(rc_register_physical_path(static_cast<RcComm*>(conn), &input, &json_error_string) == SYS_NO_API_PRIV);
+    CHECK(json_error_string == nullptr);
+}
+
+TEST_CASE("api plugin")
+{
+    load_client_api_plugins();
+    irods::experimental::client_connection conn{irods::experimental::defer_authentication};
+    CHECK(rc_touch(static_cast<RcComm*>(conn), "") == SYS_NO_API_PRIV);
+}
```
Comment (korydraughn) [inline unit_tests/src/test_rc_data_obj.cpp:1552]: How do we feel about these test cases living here?

NGL, I really didn't want to create two additional files to test these cases. If others feel that is incorrect, then I'll create the files.

Thoughts?

Comment (alanking) [reply unit_tests/src/test_rc_data_obj.cpp:1552]: This file is becoming a bit of a dumping ground anyway, so it seems fine here to me. I would request that the test case names be a little more descriptive and/or a comment indicating the purpose, please :)

Comment (korydraughn) [reply unit_tests/src/test_rc_data_obj.cpp:1552]: Agreed. Those names are placeholders until I figure out how to describe what they're about.

Comment (korydraughn) [reply unit_tests/src/test_rc_data_obj.cpp:1552]: I still have to write an issue for this PR too.

**Comment by korydraughn** [comment]
All unit tests pass.

Running core tests.

**Comment by korydraughn** [comment]
All core tests passed.

Running federation tests.

**Comment by korydraughn** [comment]
All federation tests passed.

**Review by alanking — APPROVED** [review]
Seems good to me. # when ready

**Comment by korydraughn** [comment]
#'d

**Pull request merged**